android/src/main/java/com/swmansion/rnscreens/bottomsheet/BottomSheetDialogRootView.kt

```diff
@@ -0,0 +1,102 @@
+
+ @SuppressLint("ViewConstructor")
+ class BottomSheetDialogRootView(
+    val reactContext: ReactContext?,
+    private val eventDispatcher: EventDispatcher
+ ) : ReactViewGroup(reactContext), RootView {
+    private val jsTouchDispatcher: JSTouchDispatcher = JSTouchDispatcher(this)
+    private var jsPointerDispatcher: JSPointerDispatcher? = null
+
+    init {
+        // Can we safely use ReactFeatureFlags?
+        if (ReactFeatureFlags.dispatchPointerEvents) {
+            jsPointerDispatcher = JSPointerDispatcher(this)
+        }
+    }
+
+    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
+        if (changed) {
+            // This view is used right now only in ScreenModalFragment, where it is injected
+            // to view hierarchy as a parent of a Screen.
+            if (childCount > 1) {
+                val grabberWidthSpec = MeasureSpec.makeMeasureSpec(measuredWidth, MeasureSpec.EXACTLY)
+                val grabberHeightSpec = MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED)
+
+                val grabber = getChildAt(0)
+                grabber.measure(grabberWidthSpec, grabberHeightSpec)
+                grabber.layout(l, t, r, grabber.measuredHeight)
+                getChildAt(childCount - 1).layout(l, t + grabber.measuredHeight, r, b)
+            } else {
+                getChildAt(childCount - 1).layout(l, t, r, b)
+            }
+        }
+    }
+
+    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
+        jsTouchDispatcher.handleTouchEvent(event, eventDispatcher)
+        jsPointerDispatcher?.handleMotionEvent(event, eventDispatcher, true)
+        return super.onInterceptTouchEvent(event)
+    }
+
+    override fun onTouchEvent(event: MotionEvent): Boolean {
+        jsTouchDispatcher.handleTouchEvent(event, eventDispatcher)
+        jsPointerDispatcher?.handleMotionEvent(event, eventDispatcher, false)
+        super.onTouchEvent(event)
+        return true;
+    }
+
+    override fun onInterceptHoverEvent(event: MotionEvent): Boolean {
+        jsPointerDispatcher?.handleMotionEvent(event, eventDispatcher, true)
+        // This is how DialogRootViewGroup implements this, it might be a copy-paste mistake
+        // on their side.
+        return super.onHoverEvent(event)
+    }
+
+    override fun onHoverEvent(event: MotionEvent): Boolean {
+        jsPointerDispatcher?.handleMotionEvent(event, eventDispatcher, false)
+        return super.onHoverEvent(event)
+    }
+
+
+    override fun onChildStartedNativeGesture(view: View, event: MotionEvent) {
+        jsTouchDispatcher.onChildStartedNativeGesture(event, eventDispatcher)
+        jsPointerDispatcher?.onChildStartedNativeGesture(view, event, eventDispatcher)
+    }
+
+    @Deprecated("Deprecated by React Native")
+    override fun onChildStartedNativeGesture(event: MotionEvent) {
+        throw IllegalStateException("Deprecated onChildStartedNativeGesture was called")
+    }
```

Comment:
What do you mean by

> React Native should take this responsibility

?

I'm throwing exception here, because it is unexpected that this method is called. 

Unresolve if you want to discuss further here, please.